File: LeetcodeProblems Py/1684.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        allowed_s = set(allowed)
        ans = 0
        for word in words:
            good = True
            for c in word:
                if c not in allowed_s:
                    good = False 
                    break
            if good:
                ans += 1
        return ans

File: LeetcodeProblems Cpp/1684.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {

        // make set for input string 
        std::unordered_set<char> set(allowed.begin(), allowed.end());
        int ans = 0;
        for (auto word : words) {
            bool flag = true;
            for (char c : word) {
                if (set.find(c) == set.end()) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                ans++;
            }
        }

        return ans;
    }
};